src/components/auth/Login.jsx

```diff
@@ -0,0 +1,106 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { useAuth } from "../../context/AuthContext";
+import { validLogin } from "../../utilities/auth-utils";
+
+const Login = ({ toggleAuth }) => {
+  const { setAuth } = useAuth();
+  const [login, setLogin] = useState({
+    email: "",
+    password: "",
+  });
+
+  const [loginErrors, setLoginErrors] = useState({
+    email: "",
+    password: "",
+    others: "",
+  });
+
+  const loginInputHandler = (e) => {
+    setLogin((data) => ({ ...data, [e.target.name]: e.target.value }));
+    setLoginErrors((err) => ({ ...err, [e.target.name]: "" }));
+  };
+
+  const handleLoginSubmit = (e) => {
+    e.preventDefault();
+    const { isValid, errors } = validLogin(login, loginErrors);
+
+    if (!isValid) {
+      setLoginErrors(errors);
+      return;
+    }
+
+    postLoginData(login.email, login.password);
+  };
+
+  const postLoginData = async (email, password) => {
+    try {
+      const { data, status } = await axios.post("/api/auth/login", {
+        email,
+        password,
+      });
+      if (status !== 200) return;
+      setAuth({ isLoggedIn: true, encodedToken: data.encodedToken });
+    } catch (err) {
+      console.log(err.response);
+    }
+  };
+  const testLogin = () => {
+    setLogin({ email: "adarshbalika@gmail.com", password: "adarshBalika123" });
+  };
+
+  return (
+    <form className="flex-cl h30 fixed-w30" onSubmit={handleLoginSubmit}>
+      <div className="border-bottom">
+        <button className=" w50 border-rg font-lg pd">LOG IN</button>
+        <button
+          className=" w50 font-lg border-reset pd"
+          onClick={() => toggleAuth(false)}
+          type="button"
+        >
+          SIGNUP
+        </button>
+      </div>
+      <input
+        type="text"
+        placeholder="EMAIL"
+        name="email"
+        className="pd mg-rl-2 mg-top-3"
+        onChange={(e) => loginInputHandler(e)}
```

Comment:
you can directly give `loginInputHandler` instead of `(e) => loginInputHandler(e)`

Reply:
Yes you are right , I will make the changes